### PR TITLE
Tray Notifier updates

### PR DESCRIPTION
**Pull request by roxanneskelly**
https://highfidelity.fogbugz.com/f/cases/18791/Tray-Notifier-updates

parent is https://highfidelity.manuscript.com/f/cases/18504

Add notifications to the server-console system tray app for featured stories, connection login, economic activity, and marketplace updates.

You'll need to run the server + client build, and install the server

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13013/HighFidelity-Beta-PR14002-f419937.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13013/HighFidelity-Beta-PR14002-f419937.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13013/HighFidelity-Beta-Interface-PR14002-f419937.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13016/HighFidelity-Beta-PR14002-555432d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13016/HighFidelity-Beta-PR14002-555432d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13016/HighFidelity-Beta-Interface-PR14002-555432d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13016/HighFidelity-Beta-PR14002-555432d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13016/HighFidelity-Beta-Interface-PR14002-555432d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13103/HighFidelity-Beta-PR14002-d5f756c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13103/HighFidelity-Beta-PR14002-d5f756c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13103/HighFidelity-Beta-Interface-PR14002-d5f756c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13106/HighFidelity-Beta-PR14002-aba183b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13106/HighFidelity-Beta-PR14002-aba183b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13106/HighFidelity-Beta-Interface-PR14002-aba183b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13118/HighFidelity-Beta-PR14002-bda1796.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13118/HighFidelity-Beta-PR14002-bda1796.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13118/HighFidelity-Beta-Interface-PR14002-bda1796.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13118/HighFidelity-Beta-PR14002-bda1796.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13118/HighFidelity-Beta-Interface-PR14002-bda1796.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13157/HighFidelity-Beta-PR14002-6e051b7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13157/HighFidelity-Beta-PR14002-6e051b7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13157/HighFidelity-Beta-Interface-PR14002-6e051b7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13157/HighFidelity-Beta-PR14002-6e051b7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13157/HighFidelity-Beta-Interface-PR14002-6e051b7.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13183/HighFidelity-Beta-PR14002-61cbee7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13183/HighFidelity-Beta-PR14002-61cbee7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13183/HighFidelity-Beta-Interface-PR14002-61cbee7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13183/HighFidelity-Beta-PR14002-61cbee7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13183/HighFidelity-Beta-Interface-PR14002-61cbee7.exe

**Comment by howard-stearns** [comment]
Nice! Got messages, got light. Was able to click on notice to go to that person (actually, there was a protocol mismatch so I could not completely test for sure), and was able to click on a console items to open interface (if necessary) and open the corresponding app.

Some punchlist issues:
- Enable notifications settings isn't persisted across restarts.
- What to do if user isn't signed in, and therefore no AccountInfo.bin?
- I got confused as to when it was checking. E.g., should toggling notifications  off and on do the "startup" poll? We'll also have to sync your code and zach's, and see if the messages feel right when you start up Interface as a result of the actions.

File: server-console/src/modules/hf-process.js [hunk 1]
```diff
@@ -259,6 +259,24 @@ Process.prototype = extend(Process.prototype, {
...
+                    var exeData = line.split(",");
+                    var executable = exeData[0].replace(/\"/g, "").toLowerCase();
+                    if(executable == _command) {
+                        running = true;
+                    }
+                });
+                done(running);
+            });
+        } else if (os.type == 'Darwin') {
+            console.log("TODO IsRunning Darwin");
```
[the remark this answers is not in the document]
Comment (roxanneskelly) [reply server-console/src/modules/hf-process.js]: Yeah, this will be a separate PR

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13249/HighFidelity-Beta-PR14002-6f4c376.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13249/HighFidelity-Beta-Interface-PR14002-6f4c376.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13249/HighFidelity-Beta-PR14002-6f4c376.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13249/HighFidelity-Beta-Interface-PR14002-6f4c376.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13264/HighFidelity-Beta-PR14002-c1d2f00.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13264/HighFidelity-Beta-PR14002-c1d2f00.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13264/HighFidelity-Beta-Interface-PR14002-c1d2f00.exe

**Comment by roxanneskelly** [comment]
Install build and DON'T configure console to run on startup

Make sure you have some wallet transactions for your primary user.
Create a marketplace item, put it on the marketplace, and purchase it with your primary user
Create a featured story (https://staging.highfidelity.com/user_stories/new)
Log in with a secondary user on a separate machine.


In a cmd.exe window, type:
'cd c:\program files\<hifi install dir>\server-console'
'set HIFI_METAVERSE_URL=https://staging.highfidelity.come'
'server.console.exe'

Validate notifications come up for users, stories, wallet, and marketplace

Open server console menu

Validate all 4 notification types are 'lit'

Click on GoTo.  Validate interface starts and GoTo app comes up.
Validate GoTo light goes out
Stop interface

Click People.  Validate interface starts.  Validate People app comes up
Validate People light on menu goes out.
Stop interface

Click Wallet.  Validate interface starts.  Validate Wallet app comes up.
Validate Wallet light goes out
Stop interface

Click Market. Validate interface comes up.  Validate Market app comes up.
Validate Market light goes out.

Keep interface open.

Select Goto, show GoTo app comes up
Select People, show People app comes up
Select Wallet, show Wallet app comes up
Select Market, show Market app comes up

Kill interface

Kill 2nd interface

Wait 2 minutes 30 seconds

Start 2nd interface

Validate notification of user login comes up (may take upwards of 2 minutes)
Click on notification.
Validate interface starts and you show up next to 2nd user
kill interface

Click people notification
Validate interface starts.  Validate we rez next to 2nd user.
Close interface.

Create new Featured Story (or blast by second user)
Validate featured story notification comes up (may take up to 2 minutes)
Click on featured story notification
Validate interface starts.  Validate rez in featured story location.
Kill interface.

(do the following twice, both with memo and without)
2nd user gives first user HFC
Validate wallet notification comes up (up to 10 minutes)
Click on wallet notification.
Validate interface starts.  Validate wallet up comes up.

(do the following twice, both with memo and without)
2nd user gives Gift to primary user
Validate wallet notification comes up (up to 10 minutes)
Click on wallet notification.
Validate interface starts.  Validate wallet up comes up.

2nd user purchases marketplace item created by first user.
Validate wallet notification comes up (up to 10 minutes)
Click on wallet notification.
Validate interface starts.  Validate wallet up comes up.

Create update for marketplace item owned by first user
Validate marketplace notification comes up
Click on marketplace notification.
Validate interface starts.  Validate marketplace app comes up.
kill interface
kill interface for 2nd user

Wait 2 minutes

Disable notifications via menu
Validate notification types disappear from menu
Log in 2nd user
Validate there is no notification (can take upwards of 2.5 minutes)
Create featured story.
Validate there is no notification (can take upwards of 2.5 minutes)
Create update for marketplace item.
Validate there is no notification (can take upwards of 10.5 minutes)
Give HFC from 2nd user to the first.
Validate there is no notification (can take upwards of 10.5 minutes to poll)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13330/HighFidelity-Beta-PR14002-3deb3bc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13330/HighFidelity-Beta-PR14002-3deb3bc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13330/HighFidelity-Beta-Interface-PR14002-3deb3bc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13330/HighFidelity-Beta-PR14002-3deb3bc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13330/HighFidelity-Beta-Interface-PR14002-3deb3bc.exe

**Comment by rickcarr044** [comment]
Currently testing PR14002

**Comment by rickcarr044** [comment]
On desktop after install - the glyph on the toolbar's GOTO icon is missing.

gotoglyphmissing

**Comment by rickcarr044** [comment]
PASS @conklin94122

Ran through the steps in the test plan and it all behaved (Thanks Roxie for the assistance and patience).  

- Confirmed text notifications appeared in Console in the System Tray for GOTO, PEOPLE, WALLET and MARKET for set events.
- Confirmed clicking on the notifications opened Interface or to open an app if Interface is already open
- Confirmed functionality with an assortment of Featured Stories, Snap Blasts, Connection offline/online, Sending Money and giving Gifts that triggered the Systray notifications.
- Confirmed the links between the notifications and the user's app and client 2 domain

**Comment by conklin94122** [comment]
@roxanneskelly I'm removing QA Approved due to requested dev changes. If those changes do not require retesting let me know. Otherwise relabel as QA Ready when the changes are approved.

**Comment by roxanneskelly** [comment]
Test plan for merge conflict fix:
* Bring up interface
* Click on 'GoTo' in tray menu
* Verify the GOTO app comes up in the running interface.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13405/HighFidelity-Beta-PR14002-af6db93.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13405/HighFidelity-Beta-PR14002-af6db93.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13405/HighFidelity-Beta-Interface-PR14002-af6db93.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13407/HighFidelity-Beta-PR14002-eec565e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13407/HighFidelity-Beta-PR14002-eec565e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13407/HighFidelity-Beta-Interface-PR14002-eec565e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13407/HighFidelity-Beta-PR14002-eec565e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13407/HighFidelity-Beta-Interface-PR14002-eec565e.exe

**Comment by rickcarr044** [comment]
Three of the groups behave as expected, but the PEOPLE app displays a slightly different behavior in the current build as opposed to yesterday's build.  

With yesterday's build (PR14002-3deb3bc) when Client B went from Offline availability to Friends and Connections it triggered PEOPLE notifications in the systray (Text banner and red dot next to People item).  In the same scenario with today's build (PR14002-eec565e) it does not produce PEOPLE notifications.  (However it does produce Toolbar PEOPLE's notifications - see attached screenshots).  Writing a ticket in Manuscript and will update this ticket when completed.

**Comment by rickcarr044** [comment]
Ran a test again and this time People notifications did appear in the system tray - probably related to not allocating sufficient test time moving from F&C availability to Offline then back to F&C.

**Pull request merged**